### ISSUE TITLE
Image upload always uploads the same image

### ISSUE THREAD

**Issue by cliftonts** [OpenBazaar/openbazaar-go#780]
I am slowly being driven crazy by the image upload method. I have followed the instructions in the guide:
https://api.openbazaar.org/#f2e9e0c3-be12-1e06-5321-339b662cc813

The issue is that my listing always seems to upload with the image from the godfather example. My image upload code is as follows:

```
url = "http://user:password@localhost:4002/ob/images"

	payload = "[{\n\t\"filename\": \""+fname+"\",\n\t\"image\": \""+imageData+"\"}]"
	headers = {
	    'content-type': "application/json",
	    }

	response = requests.request("POST", url, data=payload, headers=headers)
	response = json.loads(response.text)
```

The relevant chunk from my list product code is:

`payload = "{\n\t\"slug\": \""+slug+"\",\n\t\t\"termsAndConditions\": \"\",\n\t\t\"refundPolicy\": \"\",\n\t\t\"item\": {\n\t\t\t\"title\": \""+title+"\",\n\t\t\t\"description\": \""+description+"\",\n\t\t\t\"processingTime\": \"3 days\",\n\t\t\t\"tags\": [\""+tags+"\"],\n\t\t\t\"categories\": [\""+categories+"\"],\n\t\t\t\"nsfw\": true,\n\t\t\t\"condition\": \"NEW\",\n\t\t\t\"images\": [{\n\t\t\t\t\"filename\": \""+imageFilename+"\",\n\t\t\t\t\"tiny\": \""+tiny+"\",\n\t\t\t\t\"small\": \""+small+"\",\n\t\t\t\t\"medium\": \""+medium+"\",\n\t\t\t\t\"large\": \""+large+"\",\n\t\t\t\t\"original\": \""+original+"\"\n\t\t\t}],\n\t\t\t\"price\": \""+price+"\",\n\t\t\t\"options\": []\n\t\t},\n\t\t\"metadata\": {\n\t\t\t\"contractType\": \"PHYSICAL_GOOD\",\n\t\t\t\"format\": \"FIXED_PRICE\",\n\t\t\t\"expiry\": \"2037-12-31T05:00:00.000Z\",\n\t\t\t\"pricingCurrency\": \"GBP\"\n\t\t},\n\t\t\"shippingOptions\":`

**Comment by jjeffryes** [comment]
Is your imageData in base64 format? 

What do you mean by  "the godfather example"? 

The response you get from the images api is the filename and hashes for the image you uploaded. You will need to save that data as part of the POST to listing later in a 2nd step.

**Comment by cliftonts** [comment]
If you look at the example post a listing example in the link I provided you will see it creates a listing for an item called 'the godfather', the image for that listing is the image that shows up on my listing regardless of what I am listing. The guide in that link is terrible, it gives you just enough info to allow you to know what you are supposed to be doing but skips all the specifics so you can't precisely carry it out. I meant to include the code which is creating the image data but forgot.
```

with open("/home/gareth/.openbazaar2.0/root/images/original/"+row[13], "rb") as imageFile:
    		imageData = base64.b64encode(imageFile.read())
```

I'm really not sure what you mean when you say 'You will need to save that data as part of the POST to listing later in a 2nd step.' so I may or may not be doing that already.

**Comment by jjeffryes** [comment]
"http://user:password@localhost:4002/ob/images" does only one thing. It publishes the images you give it, and returns the IPFS hashes for those images, as well as the hashes for smaller versions of each image. 

That call does not associate the image with any particular use. You could use it to save an avatar or header image, for instance.

To save a listing with images, you would need to take 2 separate steps.

1. save one or more images, and retrieve the IPFS hashes for those images.
2. save the listing data, with the image data returned in step 1.

**Comment by cliftonts** [comment]
In my initial post you can see my image upload code and part of the post statement for listing the item. The postage details go on forever so I won't post  the whole thing. Within the post statement I am passing in variables for slug, title, description, price, inventory, tag, category, image name and each of the hashes. But for some reason it is ignoring that and the listing always ends up with the same image which I don't even possess. Bizarrely before I figured out what to do I was posting the image data for the godfather image (taken from the example) but changing the filename and to begin with it was listing with the correct image. This makes no sense at all.

**Comment by cliftonts** [comment]
Here is a listing showing the issue:
ob://QmRM3FuiHzYzZEMWMLxknFagxCq78LfeTT198ua5jHzs4T/store/Rocks-Off-Ass-Berry-7-Speed-Raspberry

**Comment by cliftonts** [comment]
I have located the issue! I was certain the code was correct and have scoured it many times looking for any reference to anything from the example. On a hunch I have gone back to my images directory and discovered that the images I am trying to list are actually this screenshot from the godfather. How the hell my product images have been swapped out is anybody's guess. I have just tested and all appears to be working well.
